core/src/main/scala/org/apache/spark/eventhubs/client/CachedEventHubsReceiver.scala

```diff
@@ -98,15 +98,19 @@ private[spark] object CachedEventHubsReceiver extends CachedReceiver with Loggin
 
   type MutableMap[A, B] = scala.collection.mutable.HashMap[A, B]
 
-  private[this] val receivers = new MutableMap[NameAndPartition, CachedEventHubsReceiver]()
+  private[this] val receivers = new MutableMap[String, CachedEventHubsReceiver]()
+
+  private def key(ehConf: EventHubsConf, nAndP: NameAndPartition): String = {
+    ehConf.connectionString + nAndP.partitionId
```

Comment:
a) `ehConf.connectionString + ehConf.getOperationTimeout() + ehConf.getConsumerGroup() + nAndP.partitionId`
b) ensure applying the `ReceiveTimeout` corresponding to this `EHConf` after fetching the receiver ?